File: Dsa/common/LargestSubstring.js
function LargetsSubstring(arr){
    let seen =new Set()
    let left=0
    let maxLength=0
    for(let i=0; i<arr.length; i++){
        while(seen.has(arr[i])){
            seen.delete(arr[left])
            left++
        }
        seen.add(arr[i])
        maxLength=Math.max(maxLength, i-left+1)
    }
    return maxLength
}

function CountArray(arr){
    let count=0
    for(let i=0; i<arr.length; i++){
        if(arr[i]>0){
            count++
        }
    }
    return count
}

function CountEvenOdd(arr){
    let even=0
    let odd=0
    for(let i=0; i<arr.length; i++){
        let ch=arr[i]
        if(ch%2===0){
            even++
        }
        else odd++
    }
    return {even, odd}
}

console.log(LargetsSubstring("hdsafwueasdhkjasfa"))
console.log(CountEvenOdd([2,3,4,5,6,7,6,7,8]))
console.log(CountArray([2,3,4,5,6,6,6]))

File: Dsa/common/LargestSubstring.py
def CountArray(arr):
    count=0
    for i in arr:
        count+=1
    return count

def CountEvenOdd(arr):
    even=0
    odd=0
    for i in arr:
        if i%2==0:
            even+=1
        else:
            odd+=1
    return even, odd

def LargetstringCount(arr):
    seen=set()
    left=0
    maxLength=0
    for i in range(len(arr)):
        while arr[i] in seen:
            seen.remove(arr[left])
            left+=1
        seen.add(arr[i])
        maxLength=max(maxLength, i-left+1)
    return maxLength
print(CountArray([3,4,5,6,7,7,4,5,6,7,9]))
print(CountEvenOdd([2,3,5,4,5,3,4,4,3,3,3,6,4,3]))
print(LargetstringCount("jhdajfsjdjshaajka"))
